client.go

```diff
@@ -164,12 +148,30 @@ func (c *Client) Close() error {
 	return c.Text.Close()
 }
 
+func (c *Client) greet() error {
+	// Initial greeting timeout. RFC 5321 recommends 5 minutes.
+	c.conn.SetDeadline(time.Now().Add(c.CommandTimeout))
+	defer c.conn.SetDeadline(time.Time{})
+
+	_, _, err := c.Text.ReadResponse(220)
+	if err != nil {
+		c.Text.Close()
+		if protoErr, ok := err.(*textproto.Error); ok {
+			return toSMTPErr(protoErr)
+		}
+		return err
+	}
+
+	return nil
+}
+
 // hello runs a hello exchange if needed.
 func (c *Client) hello() error {
 	if !c.didHello {
 		c.didHello = true
-		err := c.ehlo()
-		if err != nil {
+		if err := c.greet(); err != nil {
```

Comment:
don't we need to run `ehlo` independently on `greet`?

Reply:
you're right. bad code styling. better to call these steps without if/then/else.
https://levelup.gitconnected.com/how-you-can-avoid-using-else-in-your-code-871197a1adbc?gi=147a42e38aca

Reply:
`c.helloError` is not used anywhere. It doesn't make sense to store it as a `Client` member.
As well as `isTLS`:
```
$ grep isTLS client.go 
        _, isTLS := conn.(*tls.Conn)
        c.tls = isTLS
```

Reply:
`c.helloError` is used when calling `c.hello()` repeatedly.

Reply:
> c.helloError is used when calling c.hello() repeatedly

true again. but the point with `c.tls` is still valid. it is not used anywhere.

Reply:
>> c.helloError is used when calling c.hello() repeatedly

> true again

I read the code and didn't get the point for this behavior. moreover `c.helloError` is not used, when you're calling `c.ehlo` from StartTLS method.

Reply:
Just noticed another issue. When you're using 587 port usually you prefer to introduce the client side with a safe `mail.example.com` anonymous hostname (e.g. see openssl behavior: https://github.com/openssl/openssl/blob/a3e53d56831adb60d6875297b3339a4251f735d2/apps/s_client.c#L2231). And once you upgrade the connection with StartTLS you can introduce yourself with a real hostname. Currently this behavior is impossible, because the only way to set the `c.localName` is via `Hello()` method, which returns error:
https://github.com/emersion/go-smtp/blob/608f3c2840584931f2867716078c43485ea7ef3f/client.go#L186-L195

Reply:
All of these are unrelated to this PR.

Reply:
`Client.tls` dropped in 6ecbad7e0eb0.

Reply:
Opened https://github.com/emersion/go-smtp/issues/245 for the hostname thing.